plugin/trino-snowflake/src/main/java/io/trino/plugin/snowflake/SnowflakeClient.java

```diff
@@ -0,0 +1,651 @@
+                        StandardColumnMappings.varcharReadFunction(VarcharType.createUnboundedVarcharType()),
+                        StandardColumnMappings.varcharWriteFunction(),
+                        PredicatePushdownController.DISABLE_PUSHDOWN));
+            })
+            .put("variant", typeHandle -> {
+                return Optional.of(ColumnMapping.sliceMapping(
+                        VarcharType.createUnboundedVarcharType(), variantReadFunction(), StandardColumnMappings.varcharWriteFunction(),
+                        PredicatePushdownController.FULL_PUSHDOWN));
+            })
+            .put("varchar", typeHandle -> {
+                return Optional.of(varcharColumnMapping(typeHandle.getRequiredColumnSize()));
+            })
+            .put("number", typeHandle -> {
+                int decimalDigits = typeHandle.getRequiredDecimalDigits();
+                int precision = typeHandle.getRequiredColumnSize() + Math.max(-decimalDigits, 0);
+                if (precision > 38) {
+                    return Optional.empty();
+                }
+                return Optional.of(columnMappingPushdown(
+                        StandardColumnMappings.decimalColumnMapping(DecimalType.createDecimalType(
+                                precision, Math.max(decimalDigits, 0)), RoundingMode.UNNECESSARY)));
+            })
+            .buildOrThrow();
+
+    // Mappings for internal Trino types to JDBC column types
+    private static final Map<String, WriteMapping> STANDARD_WRITE_MAPPINGS = ImmutableMap.<String, WriteMapping>builder()
+            .put("BooleanType", WriteMapping.booleanMapping("boolean", StandardColumnMappings.booleanWriteFunction()))
+            .put("BigintType", WriteMapping.longMapping("number(19)", StandardColumnMappings.bigintWriteFunction()))
+            .put("IntegerType", WriteMapping.longMapping("number(10)", StandardColumnMappings.integerWriteFunction()))
+            .put("SmallintType", WriteMapping.longMapping("number(5)", StandardColumnMappings.smallintWriteFunction()))
+            .put("TinyintType", WriteMapping.longMapping("number(3)", StandardColumnMappings.tinyintWriteFunction()))
+            .put("DoubleType", WriteMapping.doubleMapping("double precision", StandardColumnMappings.doubleWriteFunction()))
+            .put("RealType", WriteMapping.longMapping("real", StandardColumnMappings.realWriteFunction()))
+            .put("VarbinaryType", WriteMapping.sliceMapping("varbinary", StandardColumnMappings.varbinaryWriteFunction()))
+            .put("DateType", WriteMapping.longMapping("date", snowFlakeDateWriter()))
+            .buildOrThrow();
+
+    private static final Map<String, WriteMappingFunction> SNOWFLAKE_WRITE_MAPPINGS = ImmutableMap.<String, WriteMappingFunction>builder()
+            .put("TimeType", type -> {
+                return WriteMapping.longMapping("time", SnowflakeClient.snowFlaketimeWriter(type));
+            })
```

Comment:
This can be a lambda expression instead of a block:

```java
.put("TimeType", type -> WriteMapping.longMapping("time", SnowflakeClient.snowFlaketimeWriter(type)))
```